File: ports/go/src/gui/layout/bcslayout.go
package layout

import "bcs/gui/widgets"

// BcsLayout governs spatial organization of UI components
type BcsLayout struct {
	children []*widgets.BcsWidget
	margin   int
	spacing  int
}

func NewBcsLayout() *BcsLayout {
	return &BcsLayout{
		children: make([]*widgets.BcsWidget, 0),
		margin:   0,
		spacing:  5,
	}
}

func (l *BcsLayout) AddWidget(w *widgets.BcsWidget) {
	l.children = append(l.children, w)
}

func (l *BcsLayout) RemoveWidget(w *widgets.BcsWidget) {
	for i, child := range l.children {
		if child == w {
			l.children = append(l.children[:i], l.children[i+1:]...)
			break
		}
	}
}

File: ports/rust/src/gui/layout/bcslayout.rs
use bcs_gui::widgets::bcswidget::BcsWidget;
use std::rc::Rc;
use std::cell::RefCell;

// BcsLayout governs spatial organization of UI components
pub struct BcsLayout {
    children: Vec<Rc<RefCell<BcsWidget>>>,
    pub margin: i32,
    pub spacing: i32,
}

impl BcsLayout {
    pub fn new() -> Self {
        BcsLayout {
            children: Vec::new(),
            margin: 0,
            spacing: 5,
        }
    }

    pub fn add_widget(&mut self, w: Rc<RefCell<BcsWidget>>) {
        self.children.push(w);
    }

    // Removing requires comparing ptrs or an ID system in Rust
    pub fn remove_widget(&mut self, _w: Rc<RefCell<BcsWidget>>) {
        // Find and remove logic
    }
}
